lib/highline/terminal/unix_stty.rb

```diff
@@ -0,0 +1,91 @@
+class HighLine
+  class Terminal::UnixStty < Terminal
+
+    # A Unix savvy method using stty to fetch the console columns, and rows.
+    # ... stty does not work in JRuby
+    def terminal_size
+      begin
+        require "io/console"
+        winsize = IO.console.winsize.reverse rescue nil
+        return winsize if winsize
+      rescue LoadError
+      end
+
+      if /solaris/ =~ RUBY_PLATFORM and
+        `stty` =~ /\brows = (\d+).*\bcolumns = (\d+)/
+        [$2, $1].map { |c| x.to_i }
+      elsif `stty size` =~ /^(\d+)\s(\d+)$/
+        [$2.to_i, $1.to_i]
+      else
+        [ 80, 24 ]
+      end
+    end
+
+                              # *WARNING*:  This requires the external "stty" program!
+    CHARACTER_MODE = "unix_stty"   # For Debugging purposes only.
+
+    def raw_no_echo_mode
+      @state = `stty -g`
+      system "stty raw -echo -icanon isig"
```

Comment:
I don't know much about stty and how different implementations exist, but currently seems you could just `include SystemExtensions::Stty` because it's exactly same for `raw_no_echo_mode` and `restore_mode` methods.